src/components/page/WithGATracker/tests/index.test.tsx

```diff
@@ -78,5 +78,22 @@ describe('components/WithGATracker', () => {
     expect(GoogleAnalytics.pageview).toBeCalledWith('/');
     history.push(PLAN_LIST_URL);
     expect(GoogleAnalytics.pageview).toBeCalledWith(PLAN_LIST_URL);
+    wrapper.unmount();
+  });
+
+  it('does not initialize GA or track page views without a GA code', () => {
+    process.env.REACT_APP_GA_CODE = '';
+    jest.mock('../../../../configs/env');
+    GoogleAnalytics.pageview = jest.fn();
+    const wrapper = mount(
+      <Provider store={store}>
+        <ConnectedRouter history={history}>
+          <App />
+        </ConnectedRouter>
+      </Provider>
+    );
+    history.push(PLAN_LIST_URL);
+    expect(GoogleAnalytics.pageview).toBeCalledTimes(0);
+    wrapper.unmount();
   });
```

Comment:
@moshthepitt @p-netm @FrankApiyo - What's the best way to set an env variable for a specific test?

The `GA_CODE` from `env.ts` is being imported into from `WithGATracker` and needs to be tested with a `GA_CODE` value of `''`.

Reply:
@cKellyDesign we typically use this file to mock them https://github.com/onaio/reveal-frontend/blob/master/src/configs/__mocks__/env.ts

Reply:
Here is an example usage in a test https://github.com/onaio/reveal-frontend/blob/b8e46731601574489ba9a4d1a679ec47885fc3a5/src/containers/forms/OrganizationSelect/tests/index.test.tsx#L16

Reply:
yeah the `jest.mock()` is there on line 86, I thought the mock file was up-to-date but perhaps it wasn't.

What I'm wondering then is how to test both cases of:
1. the `GA_CODE` is provided via env and so mounts and page tracks as normal
2. the `GA_CODE` is not provided via env and so GA tracking doesn't happen

Reply:
@cKellyDesign I think we would need to have two test files: one that mocks the envs and one that does not

Reply:
aaah ok that's a great idea, i'll do that today, thanks @moshthepitt!